### PR TITLE
Split test for tls.crt and ca.crt

### PR DESCRIPTION
**Pull request by perryan-coder**
Some new certs do not have a ca.crt so were being excluded from prometheus output. Added logging too.

For issue #53

### CODE REVIEW

File: certs_net.py [hunk 1]
```diff
@@ -8,61 +8,115 @@
...
-      x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
-      expiry = datetime.datetime.strptime(x509.get_notAfter().decode('utf-8'), "%Y%m%d%H%M%S%z").timestamp()
-      return int(expiry - time.time())
+        try:
+            cert = base64.b64decode(cert_data)
+            x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+            expiry = datetime.datetime.strptime(x509.get_notAfter().decode('utf-8'), "%Y%m%d%H%M%S%z").timestamp()
+            return int(expiry - time.time())
+        except:
+            self.logger.exception("getExpiry failed")
```
Comment (rickymoorhouse) [inline certs_net.py:44]: We should probably capture the exception detail and pass to the logger here

File: certs_net.py [hunk 2]
```diff
@@ -8,61 +8,115 @@
...
+                        self.trawler.set_gauge(
+                            'cert', 'remaining_seconds',
+                            caSecondsLeft,
+                            labels={'secret':secret.metadata.name, 'cert':'ca.crt', 'namespace': secret.metadata.namespace})
+                    if not this_secret_counted:
+                        secrets_processed += 1
+                else:
+                    self.logger.trace("No ca.crt for secret {}".format(secret.metadata.name))
+        if self.logger:
+            self.logger.info("Finished fish. Processed {}/{}".format(secrets_processed, tls_secrets_found))
```
Comment (rickymoorhouse) [inline certs_net.py:113]: Might be worth us exposing these as metrics in their own right as well so we have stats on the processing in prometheus

**Review by rickymoorhouse — CHANGES_REQUESTED** [review]
Looks good but a few changes highlighted

File: certs_net.py (hunk 1 above)
Comment (rickymoorhouse) [reply certs_net.py:44]: there's a couple of other logger.exceptions that could do with the same handling

Comment (perryan-coder) [reply certs_net.py:44]: > We should probably capture the exception detail and pass to the logger here

Doesn't logging.exception() do that anyway? What more needs to be output?
>exception(msg, *args, **kwargs)
Logs a message with level ERROR on this logger. The arguments are interpreted as for [debug()](https://docs.python.org/3/library/logging.html#logging.debug). Exception info is added to the logging message. This method should only be called from an exception handler.

Comment (rickymoorhouse) [reply certs_net.py:44]: ah I'd missed that - all good then

**Comment by perryan-coder** [comment]
Are we good to resolve the conversations and merge this, or should we perhaps firm up the exception to have error codes? If so what would they be?

**Review by rickymoorhouse — APPROVED** [review]

**Pull request merged**